drivers/interrupt_controller/Kconfig.s32

```diff
@@ -0,0 +1,11 @@
+# Configuration for NXP S32 external interrupt controller
+
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config EIRQ_S32
```

Comment:
The description of Kconfig also mentions that it is used for S32 MCUs of NXP.
Since it is an interrupt-controller, it can be renamed to something like `INTC_S32_EIRQ`.
But use `NXP` in the name of the config seems to be redundant. Beside that, in that case
we will need to update name (add NXP) some of other Kconfigs that were introduced in https://github.com/zephyrproject-rtos/zephyr/pull/49161

Reply:
The idea is that, when looking at `.config`, the reader can have some rough idea of what the kconfig is for (e.g. which subsystem). Having vendor in the kconfig would help with showing what are enabled for the board and SoC. For developers working across architectures and platforms, these would help with debugging and development.

Reply:
Okay, i will update

Reply:
I updated to use `NXP_S32_EIRQ`. Although i see there is no specific rule to add vendor in the name of Kconfig in drivers/.
Something like `INTC_S32_EIRQ` or `INTC_S32_EXTI` is enough to have a rough idea that belongs to which subsystem.